emote/memory/table.py

```diff
@@ -228,11 +266,178 @@ def _eject_count(self, count: int):
+                    f"unexpected behavior."
+                )
+
+            if "sampler_state" in config:
+                self._sampler.load_state(config["sampler_state"])
+
+            if self._length_key != config["length_key"]:
+                logger.warning(
+                    f"Deserializing memory with length key {config['length_key']}, "
+                    f"but memory is configured with length key "
+                    f"{self._length_key}. This may lead to unexpected behavior."
+                )
+
+            if self._maxlen != config["maxlen"]:
+                logger.warning(
+                    f"Deserializing memory with maxlen {config['maxlen']}, "
+                    f"but memory is configured with maxlen "
+                    f"{self._maxlen}. This may lead to unexpected behavior."
+                )
+
+            for name, column_type, column_config in config["columns"]:
+                if name not in self._columns:
+                    logger.warning(
+                        f"Deserializing memory with column {name}, "
+                        f"but memory is configured without column "
+                        f"{name}. This may lead to unexpected behavior."
+                    )
+                    continue
+
+                if column_type != self._columns[name].__class__.__name__:
+                    logger.warning(
+                        f"Deserializing memory with column {name} of type "
+                        f"{column_type}, but memory is configured with column "
+                        f"{name} of type {self._columns[name].__class__.__name__}. "
+                        f"This may lead to unexpected behavior."
+                    )
+                    continue
+
+                self._columns[name].configure(column_config)
+
+            loaded_data = {}
+            ranges = {}
+            for key in config["part_keys"]:
+                with zip_.open(f"{key}.ranges.npy", "r") as f:
+                    tuplized = np.load(f, allow_pickle=False)
+
+                    ranges[key] = {
+                        identity: (start, size) for identity, start, size in tuplized
+                    }
+
+                with zip_.open(f"{key}.npy", "r") as npz:
+                    loaded_data[key] = np.load(npz, allow_pickle=False)
+
+            # we reassemble sequences and store them in the memory
+            for identity in config["ids"]:
+                reassembled = {}
+
+                for key, data in ranges.items():
+                    (start, size) = data[identity]
+                    end = start + size
+                    reassembled[key] = loaded_data[key][start:end]
+
+                self._add_sequence_internal(identity, reassembled)
+
+            self._sampler.end_simple_import()
+            self._ejector.end_simple_import()
+
+    def _store_legacy(self, path: str) -> bool:
         """Persist the whole table and all metadata into the designated name"""
```

Comment:
Could we deprecation warn here maybe?

Reply:
Since it's now an internal function, I'm not sure if that makes sense? All existing code will be automatically upgraded to use the new format, and if you need the old format you have to explicitly opt-in.

Reply:
Added deprecation warnings.

Reply:
![image](https://github.com/EmbarkStudios/emote/assets/8234817/57623c65-043e-4c95-a4ea-3aa2fd259d28)